File: src/LCR191.gou-jian-cheng-ji-shu-zu-lcof.100338/solution.ts
function constructArr (a: number[]): number[] {
    const befores:number[] = new Array(a.length);
    let before = 1;
    for (let i = 0; i < a.length; i++) {
        befores[i] = before;
        before *= a[i];
    }
    const afters:number[] = new Array(a.length);
    let after = 1;
    for (let i = a.length - 1; i > -1; i--) {
        afters[i] = after;
        after *= a[i];
    }
    const result:number[] = new Array(a.length);
    for (let i = 0; i < a.length; i++) {
        result[i] = befores[i] * afters[i];
    }
    return result;
}

File: src/LCR191.gou-jian-cheng-ji-shu-zu-lcof.100338/solution.java
class Solution {
    public int[] statisticalResult(int[] arrayA) {
        if(arrayA.length == 0){
            return new int[]{};
        }
        int[] after = new int[arrayA.length];
        after[after.length-1] = 1;
        for(int i=after.length-2;i>-1;i--){
            after[i] = after[i+1]*arrayA[i+1];
        }
        int before = 1;
        int[] result = new int[after.length];
        for(int i=0;i< after.length;i++){
            result[i] = before*after[i];
            before *= arrayA[i];
        }

        return result;
    }
}